### ISSUE TITLE
Does LambdaLayer need BatchNorm and activation after it?

### ISSUE THREAD

**Issue by PistonY** [lucidrains/lambda-networks#12]
Hello,
I'm trying to reproduce of this.
I'm build the `LambdaResnet`,a little question is `BatchNorm and activation` are needed after this(LambdaLayer)?
Thanks.

**Comment by lucidrains** [comment]
@PistonY no I don't believe so, but feel free to correct me if I'm wrong

**Comment by PistonY** [comment]
Hi @lucidrains, thanks for reply.
I tested them all, I think you're right. When apply `bn+relu` the val accuracy doesn't grow.
This is my final [implement](https://gist.github.com/PistonY/ad33ab9e3d5f9a6a38345eb184e68cb4).

Now I'm training `LambdaResnet50`,it's looking good:
I use the standard training step in my [project](https://github.com/PistonY/ModelZoo.pytorch) same with `Resnet50` except `batch_size` set to 64. 

Epoch 28 result: Train Acc:0.44 Loss:2.54  Val Acc:0.48 Loss:3.1748e+08
Some observations are:
1. `parameters` and `GFLOPs` are small but training speed and gpu memory cost are still high.
2. Mix-Precision(FP16) training by `torch.cuda.amp` would make train loss `nan`, that's make me could only train with 64 batch_size.
3. Val loss is strange.
4. Convergence is much slower than `Resnet50`.

**Comment by lucidrains** [comment]
@PistonY Awesome, thanks for sharing! Let's keep this issue open so others can build upon it

**Comment by PistonY** [comment]
@lucidrains Unfortunately, I only got 76.1 best top1 on val set(79.2 on train set). 
I'd better wait author release their code.

**Comment by lucidrains** [comment]
@PistonY Good to know Devin, thanks for sharing!

**Comment by Kirayue** [comment]
Hi, @lucidrains 

I also build lambdaresnet50 using lambdalayer. And I encountered a situation that my loss is like PistonY's. the order is large. 

And I found that the results after lambdalayer were getting larger and larger and became nan.

here is my model code snippet for forward (just modified the [torchvision resnet](https://github.com/pytorch/vision/blob/master/torchvision/models/resnet.py#L230)  and replace the [conv2 in Bottleneck class](https://github.com/pytorch/vision/blob/master/torchvision/models/resnet.py#L113) to lambdalayer)

<img width="891" alt="Screen Shot 2020-11-02 at 3 23 21 PM" src="https://user-images.githubusercontent.com/13309212/97840636-5c05ae80-1d1f-11eb-97c5-15ebecdbe8e0.png">

the result

<img width="890" alt="Screen Shot 2020-11-02 at 3 24 02 PM" src="https://user-images.githubusercontent.com/13309212/97840688-72ac0580-1d1f-11eb-8c13-5f75a7d1fe91.png">


Could you give me some advices about this?

**Comment by zengyi-li** [comment]
> @PistonY no I don't believe so, but feel free to correct me if I'm wrong

If I understand your implementation correctly the positional Lambda interaction is purely linear, which means it should be followed by an activation and batchnorm? This is what I tried at least and its working normally.
